lib/types.nix

```
@@ -395,6 +396,35 @@ rec {
       merge = mergeEqualOption;
     };
 
+    secretFile = let
+      type = mkOptionType {
+        name = "secretFile";
+        description = "quoted path to a secret file";
+        # either the return value of makeWorldReadable or a proper string. We
+        # deliberately refuse paths, because toString x and "${x}" behave
+        # differently. Some modules might handle paths correctly, others will
+        # not, and anyway users can either quote the path or use makeWorldReadable.
+        check = x: ((isCoercibleToString x && (x.canBePublic or false)) || (isString x && !(hasPrefix "/nix/store" x))) && (hasPrefix "/" "${x}");
+        merge = mergeEqualOption;
+      };
+      makeWorldReadable = path:
+        # to make "${returnValue}" work, but still being able to store an attribute in it
+        if lib.isAttrs path then
+          path // { canBePublic = true; }
+        else
+          let res = {
+            type = "derivation";
+            name = "secretFileMadeWorldReadable";
+            outPath = "${path}";
+            outputs = [ "out" ];
+            out = res;
+            outputName = "out";
+            canBePublic = true;
```

how about something simpler:
```suggestion
            inherit path;
            __toString = r: r.path;
            canBePublic = true;
```

---

odd, it seems to work fine here.

---

indeed it works, I must have tested wrong.